csharp/rust/src/conreq.rs

```diff
@@ -0,0 +1,240 @@
+﻿use std::ffi::{c_int, c_longlong, c_uint, c_ulonglong};
+use std::ops::Add;
+use std::os::raw::c_char;
+use std::time::Duration;
+use glide_core::client;
+use glide_core::client::AuthenticationInfo;
+use crate::data::{NodeAddress, Utf8OrEmptyError};
+use crate::helpers;
+use crate::helpers::grab_str;
+
+#[repr(C)]
+pub enum EReadFromKind {
+    None = 0,
+    Primary = 1,
+    PreferReplica = 2,
+
+    // Define using ReadFrom.value
+    AZAffinity = 3,
+    // Define using ReadFrom.value
+    AZAffinityReplicasAndPrimary = 4,
+}
+
+#[repr(C)]
+pub struct ReadFrom {
+    pub kind: EReadFromKind,
+    pub value: *const c_char,
+}
+
+impl ReadFrom {
+    pub fn to_redis(self) -> Result<Option<client::ReadFrom>, Utf8OrEmptyError> {
+        Ok(match self.kind {
+            EReadFromKind::None => None,
+            EReadFromKind::Primary => Some(client::ReadFrom::Primary),
+            EReadFromKind::PreferReplica => Some(client::ReadFrom::PreferReplica),
+            EReadFromKind::AZAffinity => {
+                let value = helpers::grab_str_not_null(self.value)?;
+                Some(client::ReadFrom::AZAffinity(value))
+            }
+            EReadFromKind::AZAffinityReplicasAndPrimary => {
+                let value = helpers::grab_str_not_null(self.value)?;
+                Some(client::ReadFrom::AZAffinityReplicasAndPrimary(value))
+            }
+        })
+    }
+}
+#[repr(C)]
+pub enum EProtocolVersion {
```

Comment:
Why not to use `redis::ProtocolVersion` as is?

Reply:
`#[repr(C)]`